app/src/main/java/com/infomaniak/drive/utils/Utils.kt

```diff
@@ -286,6 +288,33 @@ object Utils {
         workManager.enqueueUniqueWork(DownloadWorker.TAG, ExistingWorkPolicy.APPEND_OR_REPLACE, downloadRequest)
     }
 
+    fun downloadAsOfflineFiles(context: Context, files: List<File>, userDrive: UserDrive = UserDrive(), onSuccess: () -> Unit) = liveData {
+        val workManager = WorkManager.getInstance(context)
+
+        if (files.any { it.isPendingOffline(context) }) workManager.cancelAllWorkByTag(BulkDownloadWorker::class.java.toString())
+        val inputData = workDataOf(
+            BulkDownloadWorker.FILE_IDS to files.map { it.id }.toIntArray(),
```

Comment:
The limitation seems to be 10KB. Even though 10KB of ids seems to be a lot of ids, we can handle the case. I see 2 possibilities:

1. Saving IDs in DB, getting a URI and then passing this one to the Worker using the `setInputData` method
2. Saving that list of IDs somewhere like a ViewModel, a Singleton class etc and then accessing it directly in the Worker

Do you see any other possibilities ?

Reply:
One issue with your solution, which is tagging as offline all selected files first and then, getting them from the worker to avoid passing a large list of IDs, is that the file list update itself as downloaded (green mark) when we mark them as offline. So we don't have any progress etc on each file.

Reply:
Getting files to put as offline using a new field in File is working great. I have also created a temporary realm migration that will need to be changed later. Tested with 50k files, it's working great. Sometimes, we have a CancellationException because of a time out but the worker is restarted and continue to download the remaining files.